### DISCUSSION TITLE
Use `dj` in folder name instead of `Various Artists`

### DISCUSSION THREAD

**Discussion by karserasl** [RogueOneEcho/caesura#86] [Ideas]
Im using a watch directory in qbittorrent, so it will be beneficial to choose where the transcoded .torrent file gets saved (maybe after --upload ?)

Also, why now we have 2x .torrent files just with .{INDEXER}. between?

And lastly, the directory names shouldnt change from the source, no matter what is the correct one.
For ex. 
> /Jan Liefhebber - 2006 - Techno Special Reckless Mix {UDC - UDCFCD05} [CD FLAC]/

should stay the same, just the codec should change
instead it produced:

> /Various Artists - Techno Special Reckless Mix [2006] [CD 320]/

**Comment by RogueOneEcho** [comment]
> Im using a watch directory in qbittorrent, so it will be beneficial to choose where the transcoded .torrent file gets saved (maybe after --upload ?)

You can use the `--copy-torrent-to`  option for the `upload` or `batch` commands to copy the `.torrent` files to an autoadd directory. See full docs: https://github.com/RogueOneEcho/caesura/blob/main/COMMANDS.md#caesura-upload

> Also, why now we have 2x .torrent files just with .{INDEXER}. between?


This is necessary to support uploading to multiple indexers (RED and OPS)
- `*.red.torrent` has the source set to `red`. 
- `*.ops.torrent` has the source set to `ops`. 
- `*.torrent` without the indexer suffix is non-specific and is used by the `transcode` command to determine if a transcode already exists. In theory it could just search for `.*.torrent` but this would be a slow search process if you have a lot of files in the output directory.

> And lastly, the directory names shouldnt change from the source, no matter what is the correct one. For ex.
> 
> > /Jan Liefhebber - 2006 - Techno Special Reckless Mix {UDC - UDCFCD05} [CD FLAC]/
> 
> should stay the same, just the codec should change instead it produced:
> 
> > /Various Artists - Techno Special Reckless Mix [2006] [CD 320]/

What is your reasoning for this? It would be incredibly difficult to reliably preserve the original folder name because there is no consistent naming convention.

If you're transcoding to MP3 for example you'd have to remove any reference to 24 Bit, 16 Bit, Flac, Lossless etc from the folder name.

In your example (RED 5155020) I agree that "Various Artists" isn't ideal. One possible solution would be to add some logic that if  the torrent has multiple `artists` but only one `dj` then use that as the folder name. Is this a better solution?

```
/Jan Liefhebber - Techno Special Reckless Mix [2006] [CD 320]/
```

**Reply by RogueOneEcho** [reply]
If only two artists then include both: RED 4882068, RED 5071539 RED 5147539

**Reply by RogueOneEcho** [reply]
This should probably be considered a breaking change. If a source has already been transcoded to the `Various Artists ...` folder then it won't be found when searching for the new name. Not a big issue but potentially frustrating.

**Reply by RogueOneEcho** [reply]
Implemented in 63f0a6b9

**Comment by karserasl** [comment]
Thanks alot for the response..

Solved the first 2 questions i had.

For the naming, true there isnt any convention for naming folders in music. Your suggestions for DJ mixes is nice..

Thanks again.
